test/daycare.c

```diff
@@ -0,0 +1,131 @@
+    u32 offspring = 0;
+    ASSUME(P_FAMILY_PIKACHU == TRUE);
+    ASSUME(P_GEN_2_CROSS_EVOS == TRUE);
+    ASSUME(P_FAMILY_RIOLU == TRUE);
+
+    ZeroPlayerPartyMons();
+    PARAMETRIZE { offspring = SPECIES_RIOLU; RUN_OVERWORLD_SCRIPT(givemon SPECIES_PIKACHU, 100, gender=MON_MALE;   givemon SPECIES_LUCARIO, 100, gender=MON_FEMALE;); }
+    PARAMETRIZE { offspring = SPECIES_PICHU; RUN_OVERWORLD_SCRIPT(givemon SPECIES_PIKACHU, 100, gender=MON_FEMALE; givemon SPECIES_LUCARIO, 100, gender=MON_MALE;); }
+    STORE_IN_DAYCARE_AND_GET_EGG();
+
+    EXPECT_EQ(GetMonData(&gPlayerParty[0], MON_DATA_SPECIES), offspring);
+}
+
+TEST("(Daycare) Pokémon can breed with Ditto if they don't belong to the Ditto or No Eggs Discovered group")
+{
+    u32 j = 0;
+    u32 parentSpecies = 0;
+
+    ZeroPlayerPartyMons();
+    for (j = 1; j < NUM_SPECIES; j++)
+        PARAMETRIZE { parentSpecies = j; }
+    VarSet(VAR_TEMP_C, parentSpecies);
+    RUN_OVERWORLD_SCRIPT(
+        givemon SPECIES_DITTO, 100; givemon VAR_TEMP_C, 100;
+    );
+    STORE_IN_DAYCARE_AND_GET_EGG();
+
+    if (gSpeciesInfo[parentSpecies].eggGroups[0] != EGG_GROUP_NO_EGGS_DISCOVERED
+     && gSpeciesInfo[parentSpecies].eggGroups[0] != EGG_GROUP_DITTO)
+        EXPECT_NE(GetMonData(&gPlayerParty[0], MON_DATA_SPECIES), SPECIES_NONE);
+    else
+        EXPECT_EQ(GetMonData(&gPlayerParty[0], MON_DATA_SPECIES), SPECIES_NONE);
+}
+
+TEST("(Daycare) Shellos' form is always based on the mother's form")
+{
+    u32 offspring = 0;
+    ASSUME(P_FAMILY_MEOWTH == TRUE);
+    ASSUME(P_ALOLAN_FORMS == TRUE);
+    ASSUME(P_GALARIAN_FORMS == TRUE);
+
+    ZeroPlayerPartyMons();
+    PARAMETRIZE { offspring = SPECIES_SHELLOS_WEST; RUN_OVERWORLD_SCRIPT(givemon SPECIES_SHELLOS_EAST, 1, gender=MON_MALE; givemon SPECIES_SHELLOS_WEST, 1, gender=MON_FEMALE;); }
+    PARAMETRIZE { offspring = SPECIES_SHELLOS_WEST; RUN_OVERWORLD_SCRIPT(givemon SPECIES_SHELLOS_EAST, 1, gender=MON_MALE, item=ITEM_EVERSTONE; givemon SPECIES_SHELLOS_WEST, 1, gender=MON_FEMALE;); }
+    PARAMETRIZE { offspring = SPECIES_SHELLOS_WEST; RUN_OVERWORLD_SCRIPT(givemon SPECIES_SHELLOS_EAST, 1, gender=MON_MALE; givemon SPECIES_SHELLOS_WEST, 1, gender=MON_FEMALE, item=ITEM_EVERSTONE;); }
+    PARAMETRIZE { offspring = SPECIES_SHELLOS_EAST; RUN_OVERWORLD_SCRIPT(givemon SPECIES_SHELLOS_WEST, 1, gender=MON_MALE; givemon SPECIES_SHELLOS_EAST, 1, gender=MON_FEMALE;); }
+    PARAMETRIZE { offspring = SPECIES_SHELLOS_EAST; RUN_OVERWORLD_SCRIPT(givemon SPECIES_SHELLOS_WEST, 1, gender=MON_MALE, item=ITEM_EVERSTONE; givemon SPECIES_SHELLOS_EAST, 1, gender=MON_FEMALE;); }
+    PARAMETRIZE { offspring = SPECIES_SHELLOS_EAST; RUN_OVERWORLD_SCRIPT(givemon SPECIES_SHELLOS_WEST, 1, gender=MON_MALE; givemon SPECIES_SHELLOS_EAST, 1, gender=MON_FEMALE, item=ITEM_EVERSTONE;); }
+    STORE_IN_DAYCARE_AND_GET_EGG();
+
+    EXPECT_EQ(GetMonData(&gPlayerParty[0], MON_DATA_SPECIES), offspring);
+}
+
+TEST("(Daycare) Pokémon with regional forms give the correct offspring")
```

Comment:
This test is missing Paldean checks (Paldean Wooper)

Reply:
And Hisui. I'll get to it in a bit